src/TodoList.js

```diff
@@ -0,0 +1,76 @@
+import {KEY} from './constants.js';
+
+export default function TodoList($todoList, data, {removeItem, editItem}) {
+  this.$todoList = $todoList;
+  this.data = data;
+
+  this.updateItem = (nextData) => {
+    this.data = [...nextData];
+    this.render();
+    this.bindEvents();
+  };
+
+  this.bindEvents = () => {
+    document.querySelectorAll('.todo-item').forEach(($item) => {
+      $item.addEventListener('click', ({target}) => {
+        if (target.tagName === 'INPUT' && target.classList.contains('toggle')) {
+          const $todoItem = target.closest('.todo-item');
+          const {index} = $todoItem.dataset;
+
+          if ($todoItem.classList.contains('completed')) {
+            $todoItem.classList.remove('completed');
+            this.data[index].isCompleted = false;
+          } else {
+            $todoItem.classList.add('completed');
+            this.data[index].isCompleted = true;
+          }
+        }
+
+        if (target.classList.contains('destroy')) {
+          const {index} = target.closest('.todo-item').dataset;
+          removeItem(index);
+        }
+      });
+
+      $item.addEventListener('dblclick', ({target}) => {
+        const $todoItem = target.closest('.todo-item');
+        const {index} = target.closest('.todo-item').dataset;
+        const oldValue = target.innerText;
+
+        $todoItem.classList.add('editing');
+        $todoItem.querySelector('.edit').focus();
+        $todoItem.querySelector('.edit').setSelectionRange(oldValue.length, oldValue.length);
+        $todoItem.addEventListener('keyup', ({key, target}) => {
+          if (key === KEY.ESC) {
+            target.value = oldValue;
+            $todoItem.classList.remove('editing');
+          } else if (key === KEY.ENTER) {
+            editItem(index, target.value);
+          }
+        });
+      });
+
+      $item.addEventListener('focusout', ({target}) => {
+        target.closest('.todo-item').classList.remove('editing');
+      });
+    });
+  };
+
+  this.render = () => {
+    let result = '';
+    this.data.map(({text, isCompleted}, index) => {
+      result += `
+      <li class="todo-item ${isCompleted? 'completed' : ''}" data-index="${index}">
+        <div class="view">
+          <input class="toggle" type="checkbox" ${isCompleted? 'checked' : ''} />
+          <label class="label">${text}</label>
+          <button class="destroy"></button>
+        </div>
+        <input class="edit" value="${text}" />
+      </li>
+      `;
+    }).join('');
+
+    this.$todoList.innerHTML = result;
```

Comment:
```suggestion
    this.$todoList.innerHTML = this.data.map(({text, isCompleted}, index) => `
      <li class="todo-item ${isCompleted? 'completed' : ''}" data-index="${index}">
        <div class="view">
          <input class="toggle" type="checkbox" ${isCompleted? 'checked' : ''} />
          <label class="label">${text}</label>
          <button class="destroy"></button>
        </div>
        <input class="edit" value="${text}" />
      </li>
      `).join('');;
```
이 부분도 앞서 언급한 것 처럼 이렇게 사용할 수 있습니다!